homeassistant/components/matter/lock.py

```diff
@@ -0,0 +1,131 @@
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Matter lock from Config Entry."""
+    matter = get_matter(hass)
+    matter.register_platform_handler(Platform.LOCK, async_add_entities)
+
+
+class MatterLock(MatterEntity, LockEntity):
+    """Representation of a Matter lock."""
+
+    features: int | None = None
+
+    @property
+    def supports_door_position_sensor(self) -> bool:
+        """Return True if the lock supports door position sensor."""
+        if self.features is None:
+            return False
+
+        return bool(self.features & DoorLockFeature.kDoorPositionSensor)
+
+    async def send_device_command(
+        self,
+        command: clusters.ClusterCommand,
+        timed_request_timeout_ms: int = 1000,
+    ) -> None:
+        """Send a command to the device."""
+        await self.matter_client.send_device_command(
+            node_id=self._endpoint.node.node_id,
+            endpoint_id=self._endpoint.endpoint_id,
+            command=command,
+            timed_request_timeout_ms=timed_request_timeout_ms,
+        )
+
+    async def async_lock(self, **kwargs: Any) -> None:
+        """Lock the lock with pin if needed."""
+
+        await self.send_device_command(command=clusters.DoorLock.Commands.LockDoor())
+
+    async def async_unlock(self, **kwargs: Any) -> None:
+        """Unlock the lock with pin if needed."""
+
+        await self.send_device_command(command=clusters.DoorLock.Commands.UnlockDoor())
+
+    @callback
+    def _update_from_device(self) -> None:
+        """Update the entity from the device."""
+
+        if self.features is None:
+            self.features = int(
+                self.get_matter_attribute_value(clusters.DoorLock.Attributes.FeatureMap)
+            )
+
+        lock_state = self.get_matter_attribute_value(
+            clusters.DoorLock.Attributes.LockState
+        )
+
+        assert lock_state is not None
+
+        LOGGER.debug("Lock state: %s for %s", lock_state, self.entity_id)
+
+        if lock_state is clusters.DoorLock.Enums.DlLockState.kLocked:
+            self._attr_is_locked = True
+        elif lock_state is clusters.DoorLock.Enums.DlLockState.kUnlocked:
+            self._attr_is_locked = False
+        else:
+            self._attr_is_locked = None
+            LOGGER.warning("Unknown lock state: %s for %s", lock_state, self.entity_id)
+
+        if self.supports_door_position_sensor:
+            door_state = self.get_matter_attribute_value(
+                clusters.DoorLock.Attributes.DoorState
+            )
+
+            assert door_state is not None
+
+            LOGGER.debug("Door state: %s for %s", door_state, self.entity_id)
+
+            self._attr_is_jammed = (
+                door_state is clusters.DoorLock.Enums.DlDoorState.kDoorJammed
+            )
+
+
+class DoorLockFeature(IntFlag):
+    """Temp enum that represents the features of a door lock, should be replaced by the library provided one once that is released."""
```

Comment:
Please break long strings and comments around 88 characters per line at minimum. Docstrings should not be longer than 72 characters per line according to PEP8.

```suggestion
    """Temp enum that represents the features of a door lock.

    Should be replaced by the library provided one once that is released.
    """
```